var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -166,6 +157,27 @@ def setup_environment(self, spack_env, run_env):
 
         run_env.prepend_path('CPATH', include_path)
 
+    # Do the usual with gcc
+    def get_phases(self):
+        self.phases = ['configure', 'build', 'install']
+
+    # as per https://docs.scipy.org/doc/scipy/reference/building/linux.html
+    # build and install in one step
+    @when('%intel')
+    def get_phases(self):
+        self.phases = ['install']
+
+    def install(self, spec, prefix):
+        install_args = self.install_args(self, prefix)
+        self.setup_py(*install_args)
+
+    @when('%intel')
+    def install(self, spec, prefix):
+        install_args = self.install_args(spec, prefix)
+        self.setup_py('config', '--compiler=intelem', 'build_clib',
+                      '--compiler=intelem', 'build_ext',
+                      '--compiler=intelem', 'install', *install_args)
```

Comment:
As I mentioned in https://github.com/spack/spack/issues/3204#issuecomment-406420601, this might have been happening because loading the intel compiler via module file also sets `MKL_ROOT` and in my case was only setting the `F77` variable and not the `F90` variable. This caused all sorts of issues that I resolved by manually editing the module file for intel compiler. Spack needs to eventually be able to work better with the intel compilers but that's something that should succeed https://github.com/spack/spack/pull/7469.